tests/wpunit/Auth/Token/MultipleTokensTest.php

```diff
@@ -0,0 +1,192 @@
+<?php declare(strict_types=1);
+
+namespace StellarWP\Uplink\Tests\Auth\Token;
+
+use StellarWP\Uplink\Auth\Token\Contracts\Token_Manager;
+use StellarWP\Uplink\Config;
+use StellarWP\Uplink\Register;
+use StellarWP\Uplink\Resources\Collection;
+use StellarWP\Uplink\Tests\TestUtils;
+use StellarWP\Uplink\Tests\UplinkTestCase;
+use StellarWP\Uplink\Uplink;
+
+final class MultipleTokensTest extends UplinkTestCase {
+
+	use TestUtils;
+
+	/**
+	 * @var Token_Manager
+	 */
+	private $token_manager;
+
+	protected function setUp(): void {
+		parent::setUp();
+
+		Config::set_token_auth_prefix('custom_');
+
+		// Run init again to reload the Token/Provider.
+		Uplink::init();
+
+		$this->token_manager = $this->container->get(Token_Manager::class);
+	}
+
+	/**
+	 * Sets up the container and returns the slug of a resource.
+	 *
+	 * @param array $resource
+	 * @return mixed
+	 */
+	public function setup_container_get_slug(array $resource) {
+		$collection = Config::get_container()->get(Collection::class);
+
+		Register::{$resource['type']}(
+			$resource['slug'],
+			$resource['name'],
+			$resource['version'],
+			$resource['path'],
+			$resource['class']
+		);
+
+		return $collection->get($resource['slug']);
+	}
+
+	/**
+	 * Get token-slug pairs from test resources.
+	 *
+	 * @return array
+	 */
+	private function getTokenSlugPairs(): array {
+		$resources = $this->get_test_resources();
+		$tokens = [];
+		$dynamicTokenPrefix = 'dynamic-token-value_';
+
+		foreach ($resources as $resource) {
+			$slug = $this->setup_container_get_slug($resource)->get_slug();
+			$tokens[$slug] = $dynamicTokenPrefix . $slug;
+		}
+
+		return $tokens;
+	}
+
+	/**
+	 * @test
+	 */
+	public function register_multiple_tokens(): void {
+		$tokens = $this->getTokenSlugPairs();
+
+		foreach ($tokens as $slug => $token) {
+			$this->assertTrue($this->token_manager->store($token, $slug));
+		}
+
+		// Retrieve all tokens and perform assertion
+		$all_tokens = $this->token_manager->get_all();
+		$this->assertSame($tokens, $all_tokens);
+
+		// Perform individual assertions for each slug
+		foreach ($tokens as $slug => $expectedToken) {
+			$retrieved_token = $this->token_manager->get($slug);
+			$this->assertSame($expectedToken, $retrieved_token);
+		}
+	}
+
+	/**
+	 * @test
+	 */
+	public function test_it_deletes_multiple_tokens(): void {
```

Comment:
Similar as above, lets remove the `test_` prefix since we have the docblock annotation